core/iwasm/common/wasm_c_api.c

```diff
@@ -4217,13 +4249,13 @@ wasm_instance_new_with_args(wasm_store_t *store, const wasm_module_t *module,
     }
 
     /* link module and imports */
-    if (imports) {
+    if (imports && imports->num_elems) {
 #if WASM_ENABLE_INTERP != 0
         if ((*module)->module_type == Wasm_Module_Bytecode) {
             import_count = MODULE_INTERP(module)->import_count;
 
-            INIT_VEC(instance->imports, wasm_extern_vec_new_uninitialized,
-                     import_count);
+            // INIT_VEC(instance->imports, wasm_extern_vec_new_uninitialized,
```

Comment:
Why not remove the two lines? May be a little strange as instance->imports doesn't exist. If to keep them, please use `/* */` to comment them as here it is C source code.